packages/core/src/anchor/ethereum/ethereum-anchor-service.ts

```diff
@@ -97,179 +95,216 @@ export default class EthereumAnchorService extends AnchorService {
-                    // the anchor request does not exist, fail and stop polling
-                    // TODO
-                    this.emit(cidDoc.docId, {
-                        cid: cidDoc.cid, status: AnchorStatus[AnchorStatus.FAILED], message: 'Request not found.'
-                    })
-                    poll = false
-                    break
-                }
-
-                const json = await response.json()
-                switch (json.status) {
-                    case "PENDING": {
-                        // just log
-                        break
-                    }
-                    case "PROCESSING": {
-                        this.emit(cidDoc.docId, { cid: cidDoc.cid, status: json.status, message: json.message })
-                        break
-                    }
-                    case "FAILED": {
-                        this.emit(cidDoc.docId, { cid: cidDoc.cid, status: json.status, message: json.message })
-                        poll = false
-                        break
-                    }
-                    case "COMPLETED": {
-                        const { anchorRecord } = json
-                        const anchorRecordCid = new CID(anchorRecord.cid.toString())
-
-                        this.emit(cidDoc.docId, {
-                            cid: cidDoc.cid, status: json.status, message: json.message, anchorRecord: anchorRecordCid
-                        })
-                        poll = false
-                        break
-                    }
-                }
-            } catch (e) {
-                // just log
-            }
+      })
+    );
+  }
+
+  /**
+   * Start polling for CidDocPair mapping
+   * @param cidDoc - CID to Doc mapping
+   * @param pollTime - Single poll timeout
+   * @param maxPollingTime - Global timeout for max polling in milliseconds
+   * @private
+   */
+  private poll(
+    cidDoc: CidDoc,
+    pollTime?: number,
+    maxPollingTime?: number
+  ): Observable<AnchorServiceResponse> {
+    const started = new Date().getTime();
+    const maxTime = started + (maxPollingTime | DEFAULT_MAX_POLL_TIME);
+    const requestUrl = [this.requestsApiEndpoint, cidDoc.cid.toString()].join(
+      "/"
+    );
+
+    return interval(DEFAULT_POLL_TIME).pipe(
+      concatMap(async () => {
+        const now = new Date().getTime();
+        if (now > maxTime) {
+          throw new Error("Exceeded max timeout");
+        } else {
+          const response = await fetch(requestUrl);
+          if (response.status === HTTP_STATUS_NOT_FOUND) {
+            throw new Error("Request not found");
+          } else {
+            const json = await response.json();
+            return this.parseResponse(cidDoc, json);
+          }
         }
+      })
+    );
+  }
+
+  /**
+   * Validate anchor proof on the chain
+   * @param anchorProof - Anchor proof instance
+   */
+  async validateChainInclusion(anchorProof: AnchorProof): Promise<void> {
```

Comment:
This is already a large and complex review.  Can we leave formatting changes out of it and do them in a dedicated review if needed?  As-is there's lots of noise in this review from tabbing changes that aren't relevant to the core change.  I don't really have strong feelings about number of spaces to indent with anyway, so I would rather just leave it as-is until we adapt an autoformatter to handle formatting for us.

Reply:
Just fyi, the autoformatter (pretty) we use in other repos enforce 2 spaces indention.

Reply:
#879